src/field.rs

```diff
@@ -267,9 +283,12 @@ pub(crate) trait FieldElementExt: FieldElement {
     /// Check if the largest number representable with `bits` bits (i.e. 2^bits - 1) is
     /// representable in this field.
     fn valid_integer_bitlength(bits: usize) -> bool {
-        if let Ok(bits_int) = Self::Integer::try_from(bits) {
-            if Self::modulus() >> bits_int != Self::Integer::from(Self::zero()) {
-                return true;
+        if bits <= 8 * Self::ENCODED_SIZE {
```

Comment:
The right-shift will panic if `bits_int` is larger than the number of bits the `Self::Integer` type uses, even tho it might be small enough to be representable in that type (hence no `try_from()` error)

Reply:
Ah, good catch!